test/fixtures/esm-exports.txt

```diff
@@ -0,0 +1,32 @@
+// Exporting declarations
+export let name1, name2/*, … */; // also var //| name1,name2
+export const name1 = 1, name2 = 2/*, … */; // also var, let //| name1,name2
+export function functionName() { /* … */ } //| functionName
+export class ClassName { /* … */ } //| ClassName
+export function* generatorFunctionName() { /* … */ } //| generatorFunctionName
+export const { name1, name2: bar } = o; //| name1,bar
+export const [ name1, name2 ] = array; //| name1,name2
+
+// Export list
+let name1, nameN; export { name1, /* …, */ nameN }; //| name1,nameN
+let variable1, variable2, nameN; export { variable1 as name1, variable2 as name2, /* …, */ nameN }; //| name1,name2,nameN
+let variable1; export { variable1 as "string name" }; //| string name
+let name1; export { name1 as default /*, … */ }; //| default
+
+// Default exports
+export default expression; //| default
+export default function functionName() { /* … */ } //| default
+export default class ClassName { /* … */ } //| default
+export default function* generatorFunctionName() { /* … */ } //| default
+export default function () { /* … */ } //| default
+export default class { /* … */ } //| default
+export default function* () { /* … */ } //| default
+
+// Aggregating modules
+export * from "module-name"; //| *
```

Comment:
A `*` expands in-context to all the names of the thing being re-exported, but we're just calling it `*` here. Do we need to identify the _actual_ names somehow or would we just consider it the responsibility of the user to patch it at the source rather than at the point of re-export? If so, what even is the function of capturing a `*` export here? Can we just mark it as no named export to see? Or do you think we _need_ to mark the `*` for some reason?

Reply:
Yeah because it's a re-export in totality, it's not particularly useful. I just included it here for completion's sake. I think it's important to support it here in case we want or need to support it later.

Reply:
Fair. More just a question than anything really. I don't think there's any _harm_ to it being there, just doesn't do the expanding so as long as we're fine with that then seems acceptable to me. 👍🏻

Reply:
Yeah we're fine with that for IITM usage, since it makes more sense to patch the original than the re-exported one.